src/components/TextInput/index.scss

```
@@ -107,4 +107,25 @@
       }
     }
   }
+
+  &__icon {
+    position: absolute;
+    height: 20px;
+    right: 10px;
+    top: 50%;
+    transform: translateY(-50%);
+    background-color: transparent;
+    border: 0;
+    opacity: 0.3;
+    cursor: pointer;
+
+    &.hidden {
```

Como faleu lá em cima, essa classe não ta seguindo o `BEM`, no caso ela precisaria estar assim

```suggestion
    &--hidden {
```